Set up for PostgreSQL function doc

Loquacity opened the pull request:
# Description

Create new top level section for PostgreSQL function doc.

To do:

- [x] @jacobprall Can you please do the needful on the web- side to get this set up in the nav for me?

# Version

Which documentation version does this PR apply to?

- [x] Latest (Default)
- [ ] Version 1.7
- [ ] Older [specify]

# Links

Fixes #[insert issue link, if any]

Loquacity committed:
Set up for PostgreSQL function doc

charislam commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+# Get started with PostgreSQL functions
+TimescaleDB is built on top of PostgreSQL. When you interact with TimescaleDB,
+you're usually using PostgreSQL functions. This guide helps you get up and
+running with the most common PostgreSQL functions that you need to use wirth
```
```suggestion
running with the most common PostgreSQL functions that you need to use with
```

charislam commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+running with the most common PostgreSQL functions that you need to use wirth
+TimescaleDB.
+
+* [`ORDER BY`][order-by] to order the results of your query
+* [`INSERT`][insert] to insert data into a hypertable
```
Would flip these. Makes sense both by logical order and alphabetical order

Loquacity replied:
```suggestion
* [`INSERT`][insert] to insert data into a hypertable
* [`ORDER BY`][order-by] to order the results of your query
```

charislam commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+* [`ORDER BY`][order-by] to order the results of your query
+* [`INSERT`][insert] to insert data into a hypertable
+
+[order-by]: /functions/order-by
+[insert]: /functions/FIXME
```
If this is a top-level section, I believe these would be in the style `/functions/:currentVersion:/FIXME`

Loquacity replied:
```suggestion
[order-by]: /functions/:currentVersion:/order-by
[insert]: /functions/:currentversion:/FIXME
```

charislam commented on functions/order-by.md:
```diff
@@ -0,0 +1,143 @@
+station location alphabetically.
+
+## Next steps
+For more information about the `ORDER BY` clause and how to use it within
+PostgreSQL. Make sure to check out [the PostgreSQL documentation][postgres-docs].
```
```suggestion
For more information about the `ORDER BY` clause and how to use it within
PostgreSQL, see [the PostgreSQL documentation][postgres-docs].
```

Loquacity replied:
```suggestion
For more information about the `ORDER BY` clause and how to use it within
PostgreSQL, see the [PostgreSQL documentation][[orderby-postgres]].
```

charislam commented on functions/order-by.md:
```diff
@@ -0,0 +1,143 @@
+-->
+
+[orderby-postgres]: https://www.postgresql.org/docs/current/queries-order.html
+[postgres-docs]: https://www.postgresql.org/docs/currentqueries-order.html
```
```suggestion
[postgres-docs]: https://www.postgresql.org/docs/current/queries-order.html
```

Loquacity replied:
```suggestion
```

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Apply suggestions from code review

Co-authored-by: Charis <26616127+charislam@users.noreply.github.com>

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

ryanbooz commented on functions/page-index/page-index.js:
```diff
@@ -0,0 +1,20 @@
+module.exports = [
+  {
+    title: "Get started with PostgreSQL functions",
```
I have a few concerns with this title. Would appreciate input from @davidkohn88 at least.

- "Functions" are a programatic term, of which Postgres supports them, but not all of the things we cover are functions. Our first example, `ORDER BY`, is not a function in the classical sense and I wouldn't expect anyone to refer to it as such. How else could we say "Postgres basics" without saying "basics"?

- I don't think we should have two "get started with" sections one after the other. The main thrust, IMO, should be getting people started with TimescaleDB. I'm not sure that a logical/great progression should be 
    * Install
    * get started with TimescaleDB
    * get started with Postgres

Loquacity replied:
Would "commands" work better than "functions"? I was under the impression that "function" was more accurate here. I struggled a lot to come up with something (anything!) other than "Getting Started" but couldn't think of anything that really worked. Any suggestions welcome!

charislam replied:
"Working with PostgreSQL" maybe? I'm also drawing a blank

Loquacity replied:
I could be convinced on that ... @ryanbooz ?

davidkohn88 replied:
Yeah, functions is highly misleading, I think you could say getting started with Postgres functionality?  Commands is also fine, but maybe doesn't convey what you want? Statements is another way of saying that (ie, I can run a `SELECT` statement or an `INSERT` statement). Query is another useful term, it's usually applied to `SELECT`s. Sometimes they're called SQL keywords (ie `FROM` or `ORDER BY` or `AS` or `IN`), some of the keywords have associated clauses (ie `FROM` clause or `ORDER BY` clause) in which case it refers to the part of the query. 

Honestly, I think something that explains this naming might be an incredibly useful concept doc and might really help with writing the rest of this stuff.

Loquacity replied:
Thanks for that, David! I'm actually leaning towards "keywords", based on this. Could "Working with PostgreSQL keywords" work?

davidkohn88 replied:
Hmmm...that still sounds like very odd phrasing to me. Mostly because you don't use the keywords, they are a class of thing that appears in a query, or a statement or a command, but I wouldn't say you could use them on their own? I don't know if that makes sense. You could say something like PostgreSQL keywords and how to use them? Or understanding PostgreSQL keywords? But in this case I'd say your phrasing just strikes me as odd, it's not like it's actually misleading like functions. 

If you want to go with your phrasing, I'd say it's less odd to say something like "Working with PostgreSQL commands" rather than keywords. 

The keywords one sort of feels like saying, working with your nose to understand the world instead of saying smell things? I don't know if that analogy was remotely useful.

Loquacity replied:
That is helpful, thanks David. I like "Working with PostgreSQL commands".

Loquacity replied:
Updated.

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

davidkohn88 commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+TimescaleDB.
+
+* [`INSERT`][insert] to insert data into a hypertable
+* [`ORDER BY`][order-by] to order the results of your query
```
I don't understand why these are the two things we've decided to add here?

Loquacity replied:
Oh, it's just the beginning of the index. This will get built out as we add more sub pages.

mirandaauhl commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+# Get started with PostgreSQL functions
```
```suggestion
# Get started with PostgreSQL
```

mirandaauhl commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+# Get started with PostgreSQL functions
+TimescaleDB is built on top of PostgreSQL. When you interact with TimescaleDB,
+you're usually using PostgreSQL functions. This guide helps you get up and
```
```suggestion
you're usually using PostgreSQL functionality. This guide helps you get up and
```

mirandaauhl commented on functions/index.md:
```diff
@@ -0,0 +1,11 @@
+# Get started with PostgreSQL functions
+TimescaleDB is built on top of PostgreSQL. When you interact with TimescaleDB,
+you're usually using PostgreSQL functions. This guide helps you get up and
+running with the most common PostgreSQL functions that you need to use with
```
```suggestion
running with the most common PostgreSQL keywords, functions, and concepts that you need to use with
```

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Apply suggestions from code review

Co-authored-by: mirandaauhl <82287545+mirandaauhl@users.noreply.github.com>

charislam reviewed (approved):
Did we decide on a final title change? Everything else looks good to me but I'm not seeing the title change :)

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity commented:
> Did we decide on a final title change? Everything else looks good to me but I'm not seeing the title change :)

I'm working on it 😅

Loquacity committed:
Title change

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
small fixes

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

charislam commented on page-index/page-index.js:
```diff
@@ -4,11 +4,14 @@ const timescaleCloudPageIndex = require("../cloud/page-index/page-index");
 const promscalePageIndex = require("../promscale/page-index/page-index");
 const installPageIndex = require("../install/page-index/page-index");
-const functionsPageIndex = require("../functions/page-index/page-index");
+const pgcommandsPageIndex = require("../pg-commands/page-index/page-index");
```
@jacobprall can correct me if I'm wrong on this, but I think this rename will require changes to the web-documentation code that made the top-level section

Loquacity replied:
I believe this has been done. @jacobprall can correct me if I'm wrong, though.

davidkohn88 commented on pg-commands/_template.md:
```diff
@@ -0,0 +1,152 @@
+ordering direction, either ascending or descending. If you don't specify the
+order direction for your columns, the default ordering is ascending.
+
+When you query time-series data, you need ordered results. Time occurs linearly
```
This feels incredibly strong.

davidkohn88 replied:
It's not necessary, and you don't always order things on time, it feels kinda weird to make this so strong here.

davidkohn88 commented on pg-commands/_template.md:
```diff
@@ -0,0 +1,152 @@
+| row_id | time | group | price | 
+| - | - | - | - |
+| 1 | 2022-01-01 | b | 261.12 | 
+| 2 | 2022-01-02 | a | 200.88 | 
```
Why is there a row_id column here?

davidkohn88 commented on pg-commands/_template.md:
````diff
@@ -0,0 +1,152 @@
+For example, if you have this raw  data:
+
+```sql
+stock_data:
````
Should we make the table and some of the data match somewhat exactly the data from the tutorial and link to the tutorial?

davidkohn88 replied:
(I think we probably should)

davidkohn88 commented on pg-commands/_template.md:
```diff
@@ -0,0 +1,152 @@
+stock_data:
+| row_id | time | group | price | 
+| - | - | - | - |
+| 1 | 2022-01-01 | b | 261.12 | 
```
This should not be group, it's highly confusing with group bys etc. we should avoid keywords for columns whenever possible. This should be called ticker or symbol or the like.

davidkohn88 commented on pg-commands/_template.md:
````diff
@@ -0,0 +1,152 @@
+specified column:
+
+```sql
+ORDER BY <column_name> <optional ASC or DESC>
````
this feels somewhat misleading and I'll note that we 
a) don't use this in a select clause it's just sort of floating out there, which maybe is good especially given the note below on other ways to use it, but also feels slightly weird and like we should acknowledge it somehow.
b) don't link to the SELECT clause documentation, when pretty much always this is going to be used with a SELECT
c) don't talk about the order of clauses anywhere in this document, which is often the biggest error and problem users have with using somethign like an order by clause with other clauses. 
d) don't mention that there are some other cases where order by clauses can be used outside of the standard ordering, two cases come up frequently: 1) window claues (ie PARTITION BY symbol ORDER BY time) and 2) in an aggregate expression, ie SELECT array_agg(symbol ORDER BY time) FROM stock_data will do something like this, and has the same syntax.

davidkohn88 commented on pg-commands/_template.md:
````diff
@@ -0,0 +1,152 @@
+```
+
+When you analyze weather station data across multiple cities, you need to order
+the station's readings by their time, and possibly location, to get the most
````
This is just the same example with a slightly different data set and doesn't really add much?

davidkohn88 commented on pg-commands/_template.md:
````diff
@@ -0,0 +1,152 @@
+```sql
+ORDER BY <column_name> <optional ASC or DESC>, <column_name> <optional ASC or DESC>, …
+```
+
````
can be

davidkohn88 commented on pg-commands/_template.md:
```diff
@@ -0,0 +1,152 @@
+<highlight type="note">
+Sorting operations are costly if indexes do not exist for the exact columns and
+ordering specified by the query. If sorting by multiple columns proves to be slow,
+look at an EXPLAIN plan for `(Sort)` nodes which indicate sorting that is taking
```
we should have a link somewhere describing what an explain plan is if we're going this basic on the explanation. It feels weird to assume that people don't know what an order by clause is but do know what an explain plan is.

davidkohn88 reviewed (commented):
Added some comments here, hope they're helpful.

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity commented:
@mirandaauhl would you like to address @davidkohn88 's comments?

davidkohn88 commented:
We can also just merge this and update the writing and work on it in a google doc or something, sorry for throwing a wrench in the process, I know this has been open for a while, and I don't want to slow things down!

Loquacity commented:
> We can also just merge this and update the writing and work on it in a google doc or something, sorry for throwing a wrench in the process, I know this has been open for a while, and I don't want to slow things down!

I think it's fine to work on the wording here, we don't want to rush to merge this until we know we have more pages coming down the pipe shortly. It's just that I didn't write it, so I thought I would let @mirandaauhl have a go before I start hacking away 😉

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

charislam commented:
Since we're receiving the first batch of pages tomorrow, what's the current state of this PR? Are we happy with the folder and section names that we've defined here? (I think Prashant mentioned a _different_ name that we'd settled on in a [Slack thread somewhere](https://iobeam.slack.com/archives/GP69EAX9B/p1656748106857989?thread_ts=1656721715.699239&cid=GP69EAX9B), do we need to change the name here to match?)

Loquacity commented:
> Since we're receiving the first batch of pages tomorrow, what's the current state of this PR? Are we happy with the folder and section names that we've defined here? (I think Prashant mentioned a _different_ name that we'd settled on in a [Slack thread somewhere](https://iobeam.slack.com/archives/GP69EAX9B/p1656748106857989?thread_ts=1656721715.699239&cid=GP69EAX9B), do we need to change the name here to match?)

Ah, I see. Let me poke at this today.

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity committed:
Change title per Prashant

Loquacity committed:
link fixes

Loquacity commented:
OK, I've updated that title per Prashant in Slack, and made a couple of little wording changes per David. I haven't made any significant changes, though, since the template is just a template, and I suspect we'll be getting a new ORDER BY at some point as well. It's really just there so we have something there to start with. I think this can probably merge as soon as we have something else to add.

charislam committed:
Rename folder

Rename PG commands folder to remove hyphen as it causes problems on the
web-documentation side.

charislam commented:
I renamed the folder as JavaScript does not like hyphens, so it was going to cause problems on the web-documentation side. Also made a new PR in web-documentation to change the folder name to match. The new section should build properly after this

https://github.com/timescale/web-documentation/pull/373

Loquacity commented:
First article is going through edits now, and I've confirmed that Docs are going to be responsible for adding it to GH. So I think we'll probably be ready to merge this is a couple of days.

Loquacity committed:
Merge branch 'latest' into concepts-templates-lana

Loquacity commented:
Killing, in light of ongoing discussion.
